java-backend/src/main/java/org/kframework/backend/java/util/RuleSourceUtil.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-2018 K Team. All Rights Reserved.
+package org.kframework.backend.java.util;
+
+import org.kframework.attributes.Location;
+import org.kframework.backend.java.kil.Rule;
+import org.kframework.utils.file.FileUtil;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Formatter;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Denis Bogdanas
+ * Created on 27-Aug-18.
+ */
+public class RuleSourceUtil {
+    private static final Map<Rule, String> cache = Collections.synchronizedMap(new HashMap<>());
+
+    public static boolean sourceShortEnough(Rule rule) {
+        Location location = rule.getLocation();
+        return location != null && location.endLine() - location.startLine() <= 20;
+    }
+
+    public static String loadSource(Rule rule) {
+        if (!cache.containsKey(rule)) {
```

Comment:
I'm fairly sure this function can return null, which I don't think is what you want.

Reply:
Nope. Null source is valid option and is properly handled in `appendRuleAndSource`.